client/src/project/new/components/Visibility.tsx

```diff
@@ -82,7 +92,7 @@ interface EditVisibilityModalConfirmationProps {
   isLoading: boolean;
   isSuccess: boolean;
   message: ReactNode;
-  visibility: Visibilities;
+  visibility?: Visibilities;
```

Comment:
This feels suspicious. The prop should always be defined I think.

```suggestion
  visibility: Visibilities;
```